### PR TITLE
Added ability to get ProcessBlock products in GenericConsumer

### LINKED ISSUE #471: Add support for ProcessBlock to GenericConsumer
See https://github.com/cms-sw/cmssw/issues/40299

Done in https://github.com/cms-sw/cmssw/pull/40424

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

This should fix the issue where the module threw an exception if another module in the job created a ProcessBlock product.

#### PR validation:

Code compiles.

fixes makortel/framework#471

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Modules/src/GenericConsumer.cc [hunk 1]
```diff
@@ -72,6 +75,13 @@ namespace edm {
                 edm::InputTag{branch.moduleLabel(), branch.productInstanceName(), branch.processName()});
           break;
 
+        case InProcess:
+          if (std::binary_search(processLabels_.begin(), processLabels_.end(), branch.moduleLabel()) or
+              (std::binary_search(processLabels_.begin(), processLabels_.end(), kWildcard) and
+               branch.className() != kPathStatus and branch.className() != kEndPathStatus))
```
Comment (makortel) [inline FWCore/Modules/src/GenericConsumer.cc:81]: Does the comparison to `kPathStatus` and `kEndPathStatus` make sense for non-Event branches? (lumi and run branches don't have it)

**Comment by makortel** [comment]
Comparison failures are in 11634.7 (#39803)

File: FWCore/Modules/src/GenericConsumer.cc (hunk 1 above)
Comment (Dr15Jones) [reply FWCore/Modules/src/GenericConsumer.cc:81]: Future proofing? :)

Comment (makortel) [reply FWCore/Modules/src/GenericConsumer.cc:81]: I'd like toe ProcessBlock, Run, and LuminosityBlock branches handled consistently in this respect (in either way); being afraid of confusion in the future).

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Modules/src/GenericConsumer.cc [hunk 2]
```diff
@@ -72,6 +75,12 @@ namespace edm {
                 edm::InputTag{branch.moduleLabel(), branch.productInstanceName(), branch.processName()});
           break;
 
+        case InProcess:
+          if (std::binary_search(processLabels_.begin(), processLabels_.end(), branch.moduleLabel()) or
+              std::binary_search(processLabels_.begin(), processLabels_.end(), kWildcard))
+            this->consumes(edm::TypeToGet{branch.unwrappedTypeID(), PRODUCT_TYPE},
```
Comment (makortel) [inline FWCore/Modules/src/GenericConsumer.cc:81]: This needs
```suggestion
            this->consumes<edm::InProcess>(edm::TypeToGet{branch.unwrappedTypeID(), PRODUCT_TYPE},
```
right? (sorry, noticed only now)

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Modules/src/GenericConsumer.cc (hunk 2 above)
Comment (Dr15Jones) [reply FWCore/Modules/src/GenericConsumer.cc]: Good catch. Done.

**Comment by makortel** [comment]
+1

Comparison differences are in 11634.7 (#39803 )

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
ping bot

**Pull request merged**